UndertaleModTool/CommunityScripts/FindObjectsWithSprite.csx

```diff
@@ -0,0 +1,87 @@
+﻿using System.Text;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UndertaleModLib.Util;
+using System.Text;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+EnsureDataLoaded();
+
+string[] spriteNames;
+ConcurrentBag<string> resultList = new();
+
+bool caseSensitive = ScriptQuestion("Case sensitive?");
+bool regexCheck = ScriptQuestion("Regex search?");
+string qText = regexCheck ? "Enter RegEx of sprite name" : "Enter sprite name(s)";
+string searchQuery = SimpleTextInput(qText, "Search box below", "", !regexCheck);
+
+if (String.IsNullOrEmpty(searchQuery) || String.IsNullOrWhiteSpace(searchQuery))
+{
+    ScriptError("Search query cannot be empty or null.");
+    return;
+}
+
+SetProgressBar(null, "Game objects", 0, Data.GameObjects.Count);
+StartUpdater();
+
+Regex searchRegex;
+if (regexCheck)
+{
+    if (caseSensitive)
+        searchRegex = new(searchQuery, RegexOptions.Compiled);
+    else
+        searchRegex = new(searchQuery, RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+    await Task.Run(() => Parallel.ForEach(Data.GameObjects, CheckObjectRegex));
+}
+else
+{
+    spriteNames = searchQuery.Split(new char[] { '\n', '\v' }, StringSplitOptions.RemoveEmptyEntries); // Shift+Enter -> \v
```

Comment:
This case should get handled in SimpleTextInput rather than in the script itself.

Reply:
Yes, I agree, will be changed.